src/Microsoft.ML.Data/Transforms/KeyToVectorTransform.cs

```diff
@@ -69,530 +75,776 @@ public bool TryUnparse(StringBuilder sb)
-
-            var info = Infos[iinfo];
-            int size = info.TypeSrc.ItemType.KeyCount;
-            Host.Assert(size > 0);
-
-            int cv = info.TypeSrc.VectorSize;
-            Host.Assert(cv >= 0);
-            Host.Assert(_types[iinfo].VectorSize == size * cv);
-
-            var getSrc = RowCursorUtils.GetVecGetterAs<uint>(NumberType.U4, input, info.Source);
-            var src = default(VBuffer<uint>);
-            return
-                (ref VBuffer<Float> dst) =>
-                {
-                    getSrc(ref src);
-                    int lenSrc = src.Length;
-                    Host.Check(lenSrc == cv || cv == 0);
-
-                    // Since we generate values in order, no need for a builder.
-                    var valuesDst = dst.Values;
-                    var indicesDst = dst.Indices;
-
-                    int lenDst = checked(size * lenSrc);
-                    int cntSrc = src.Count;
-                    if (Utils.Size(valuesDst) < cntSrc)
-                        valuesDst = new Float[cntSrc];
-                    if (Utils.Size(indicesDst) < cntSrc)
-                        indicesDst = new int[cntSrc];
-
-                    var values = src.Values;
-                    int count = 0;
-                    if (src.IsDense)
-                    {
-                        Host.Assert(lenSrc == cntSrc);
-                        for (int slot = 0; slot < cntSrc; slot++)
-                        {
-                            Host.Assert(count < cntSrc);
-                            uint key = values[slot] - 1;
-                            if (key >= (uint)size)
-                                continue;
-                            valuesDst[count] = 1;
-                            indicesDst[count++] = slot * size + (int)key;
-                        }
-                    }
-                    else
-                    {
-                        var indices = src.Indices;
-                        for (int islot = 0; islot < cntSrc; islot++)
-                        {
-                            Host.Assert(count < cntSrc);
-                            uint key = values[islot] - 1;
-                            if (key >= (uint)size)
-                                continue;
-                            valuesDst[count] = 1;
-                            indicesDst[count++] = indices[islot] * size + (int)key;
-                        }
-                    }
-                    dst = new VBuffer<Float>(lenDst, count, valuesDst, indicesDst);
-                };
+            Contracts.CheckValue(input, nameof(input));
+            return new OutVectorColumn<TKey, TValue>(input, new Config(bag));
         }
+
+        /// <summary>
+        /// Takes a column of key type of known cardinality and produces an indicator vector of floats.
+        /// </summary>
+        public static Vector<float> ToVector<TKey, TValue>(this Vector<Key<TKey, TValue>> input, bool bag = DefaultBag)
```

Comment:
>public static Vector<float> ToVector<TKey, TValue>(this Vector<Key<TKey, TValue>> input, bool bag = DefaultBag) [](start = 8, length = 111)

One of the most important usages of `KeyToVector` comes in the `VarVector` input case (e.g., text processing). It is crucially important that we support that scenario. In that case if you wanted to support bagging vs. not, you will need two overloads. I suggest to name one of them `ToVector` (no bagging, returns `VarVector<float>`), and the other to name it `ToBaggedVector` (bagging, returns `Vector<float>`).

Because of this distinction is necessary for the `VarVector` input case (which is again really important), I suggest we retain the distiction for consitency sake for the fixed size vector case, for reasons of user consistency.